compiler/rustc_codegen_llvm/src/llvm/ffi.rs

```diff
@@ -955,6 +955,26 @@ pub mod debuginfo {
             }
         }
     }
+
+    /// LLVMRustDebugNameTableKind
+    #[derive(Clone, Copy)]
+    #[repr(C)]
+    pub enum DebugNameTableKind {
+        Default,
+        None,
+        Gnu,
+    }
+
+    impl DebugNameTableKind {
+        pub fn from_generic(kind: rustc_session::config::DebugNameTableKind) -> Self {
```

Comment:
Could you avoid using `Self` in non-generic contexts, here and below?

Reply:
Okay, but why?

Reply:
`Self` in non-generic contexts almost always makes code searchability and readability worse.